### DISCUSSION TITLE
What is windows compatible font?

### DISCUSSION THREAD

**Discussion by zhaob1n** [ryanoasis/nerd-fonts#891] [Q&A] [answered]
I see all nerd fonts have a windows compatible version. What is that?

**Comment by Finii** [comment, marked as answer]
The only difference is in the font naming.

The FontName and the FontFamilyName are kept as short as possible. For examle the name contains "NF" instead of "Nerd Font" (example "Lilex NF" versus "Lilex Nerd Font"). Or "M" instead of "Mono". This is done because the names on Windows 'shall not' be longer then 32 characters.

This limit of font name length under Windows ... I believe it does not exist anymore. I always use the 'normal' fonts with long names under Windows (10), and never had any issues. I guess some old applications (like Windows 3 Word :-D) had a fixed size font chooser dialogue and could not display long names.

The Windows font specification does not mention any limit (maybe it did some years back):
Look at `Code 1` and `Code 4`:
https://docs.microsoft.com/en-us/typography/opentype/spec/name#name-ids

There is a length limitation mentioned for `Code 6` (the Postscript name), but that limit is for all platforms and is observed in all font files.

Here are some links that suggest the limit does exits:
On Windows 7: https://forum.glyphsapp.com/t/overly-strict-font-name-max-length-recommendation-in-naming-tutorial/10164
Implemented in `fontforge`: https://github.com/fontforge/fontforge/blob/582bd41a9bf04326300fc02a677fe3610d6d3ccd/fontforgeexe/savefontdlg.c#L1408
Font generation hints: https://www.fonttutorials.com/how-to-name-font-family/

**Reply by zhaob1n** [reply]
thx

**Reply by Finii** [reply]
aaronbell adds this nice tidbit of information https://github.com/ryanoasis/nerd-fonts/issues/365#issuecomment-1255467681:

> I don't think there's any documentation on it on the Microsoft typography website (it isn't really spec related). There's some info / investigation here worth reading: https://github.com/googlefonts/fontbakery/issues/2179
tldr: It is primarily a legacy issue in certain applications and situations, but appears to crop up in unexpected places, so the recommendation is to keep less than 29 characters long to avoid any problems.

**Reply by ryanoasis** [reply]
Yes, I believe I had those created based mostly on fontforge docs themselves.

**Reply by ashemedai** [reply]
@Finii I did some digging myself a while ago on this wrt VS Code and historical Windows APIs: https://github.com/notofonts/latin-greek-cyrillic/issues/9#issuecomment-1160560435
